### PR TITLE
Line graph and vector centrality functions added

### PR DESCRIPTION
**Pull request by goznalo-git**
Hi there, 

I really like this library, and I've been using it for a while now. I would like to have these two additions included, and I'm sure others would benefit from them.
- The function `convert_to_line_graph()` takes a Hypergraph and constructs its associated line graph (a pairwise graph whose nodes correspond to the edges of the Hypergraph, and two of those nodes are linked together if their associated edges share at least one node).
- The function `vector_centrality()`, defined in https://doi.org/10.1016/j.chaos.2022.112397, which allows one to extract a vector-like eigenvector centrality of a Hypergraph by using the abovementioned connection to its line graph. 

I updated the changelog and the docs, as well as making sure the formatting was fine with isort and black.

best,
Gonzalo

### CODE REVIEW

File: docs/source/api/algorithms/xgi.algorithms.centrality.rst
Comment (leotrs) [inline docs/source/api/algorithms/xgi.algorithms.centrality.rst:14]: I know this is probably the canonical name in the reference paper, but it might be too general for a general purpose library like XGI. Perhaps we should consider `line_vector_centrality` to make allusion to the fact that this uses the line graph.

File: xgi/convert.py
Comment (leotrs) [inline xgi/convert.py:140]: Tiny suggestion:
```suggestion
    """Line graph of the hypergraph.
    
    The line graph of the hypergraph `H` is the graph whose
    nodes correspond to each hyperedge in `H`, linked together
    if they share at least one vertex.
```

File: xgi/algorithms/centrality.py
Comment (leotrs) [inline xgi/algorithms/centrality.py:258]: Tiny suggestion, for uniformity with other docstrings elsewhere
```suggestion
    """The vector centrality of nodes in the line graph of the hypergraph.
```

File: xgi/algorithms/centrality.py
Comment (leotrs) [inline xgi/algorithms/centrality.py:290]: Suggestion (might be slower?):
```suggestion
    D = H.edges.size.max()
```

File: xgi/algorithms/centrality.py
Comment (leotrs) [inline xgi/algorithms/centrality.py:292]: This loop is first iterating over each possible edge order, then filtering to get all edges of said order, and the iterating over each node of each edge of those edges. So it's actually a four-ly nested loop. I wonder if there is a way to iterate over each edge (and each node in each edge) once?

File: xgi/algorithms/centrality.py
Comment (leotrs) [inline xgi/algorithms/centrality.py:302]: It's not super clear to me what is mean there by Pythonic indexing. Does this mean that the algorithm does not work for nodes that are labeled with strings? Does it work with nodes that are labeled with non-consecutive integers, or integers not starting at zero?

**Review by leotrs — COMMENTED** [review]
Dear Gonzalo, thank you for your contribution! The entire purpose of XGI from its inception was to be a tool for the community by the community, so your involvement is more than welcome. I have reviewed the code and it looks like exactly the kind of addition we are always open to receiving. I have left some comments that can hopefully be solved in a few quick iterations and we'll be sure to merge this after that.

Perhaps another pair of eyes would be useful too @nwlandry @maximelucas.

After all the comments are resolved, it would also be nice to have one or two reference tests for the new features.

Thanks again!

**Comment by maximelucas** [comment]
Thanks a lot @goznalo-git ! This looks like a quality PR.
I'll have a look this week.

File: xgi/convert.py
Comment (maximelucas) [inline xgi/convert.py:159]: Current code will give new labels to the hyperedges from 0 to M. This suggestion keeps the hyperedges labels, which might be unordered, and strings etc. Check it does not mess up the rest of your code.
```suggestion
    edge_label_dict = {tuple(edge): index for index, edge in H._edge}
```

File: xgi/convert.py
Comment (maximelucas) [inline xgi/convert.py:160]: Current code is correct but will add nodes in their order of appearances in hyperedges. The following adds them in an ordered way:
```suggestion
    nodes = sorted(set(edge_label_dict.values()))
    LG.add_nodes_from(nodes)
```

File: xgi/convert.py
Comment (maximelucas) [inline xgi/convert.py:164]: Your code definitely works. I'm wondering if it might be faster using `H.edges.neighbors()`. I give it a try below, feel free to discard.
```suggestion
    edges = []
    for edge_id1 in H.edges():
        for edge_id2 in H.edges.neighbors(edge_id1):
            edges.append({edge_id1, edge_id2})
    LG.add_edges_from(edges)
```

File: xgi/algorithms/centrality.py
Comment (maximelucas) [inline xgi/algorithms/centrality.py:285]: Do you need labeling from 0 to M? If not see below for suggestion to keep hyperedges labels

File: xgi/algorithms/centrality.py
Comment (maximelucas) [inline xgi/algorithms/centrality.py:290]: Shouldn't this go up to `D` since you're iterating over sizes?

File: xgi/algorithms/centrality.py
Comment (maximelucas) [inline xgi/algorithms/centrality.py:293]: Should also work with (check if not slower). Then you don't need to create `hyperedge_dims` anymore 
```suggestion
        for edge in H.edges.filterby("size", k).members():
```

**Comment by maximelucas** [comment]
Thank you very much @goznalo-git. I added some suggestions to make use of some xgi features.
Thanks a lot!

**Comment by goznalo-git** [comment]
Hi @leotrs @maximelucas 

Thanks for your feedback! There were some interesting suggestions and improvements, indeed. Let me know what you think of the actual status. 

best!

**Comment by maximelucas** [comment]
Thank you @goznalo-git. I'll approve this for merging now, and raise some issues for our remaining comments so we can solve later together with adding tests.

**Review by maximelucas — APPROVED** [review]

**Comment by leotrs** [comment]
@goznalo-git, thank you for your contribution. You are more than welcome to continue contributing anything you'd find useful, and to come chat with us whenever on our [zulip server](https://xgi.zulipchat.com/join/o3sjqnigvvpxma2fzv6g2m52/).

**Pull request merged**
